lib/Dialect/FIRRTL/Transforms/GrandCentral.cpp

```diff
@@ -380,7 +380,11 @@ void GrandCentralPass::runOnOperation() {
   if (annotations.empty())
     return;
 
-  auto builder = OpBuilder::atBlockEnd(circuitOp->getBlock());
+  // Setup the builder to create ops _inside the FIRRTL circuit_.  This is
+  // necessary because interfaces and interface instances are created.
+  // Instances link to their definitions via symbols and we don't want to break
+  // this.
+  auto builder = OpBuilder::atBlockEnd(circuitOp.getBody());
```

Comment:
This seems reasonable!

Reply:
Nice. There are potentially other ways of doing it, but this works fine.